### ISSUE TITLE
Ask for confirmation before deleting a task

### ISSUE THREAD

**Issue by donbex** [kdheepak/taskwarrior-tui#47]
Given that deleting a task is irreversible, would it be possible to add a dialogue to confirm the operation to prevent accidental data loss?

Something like
```
Delete [task description]?
yes   (no)
```
(where `()` denotes the preselection) should be enough.

**Comment by kdheepak** [comment]
Using `u` will undo the delete, but it would be nice to prevent accidental deletes using `d`. How about `dd` for delete instead of `d`?

**Comment by donbex** [comment]
I think `xx` would do. However, it would still be nice to have confirmation of the task being deleted. Would it be possible to echo the id and description of the task on the bottom line?

To be precise, I'm thinking that the first `x` could replace
```
╭Filter Tasks────────────────────────────╮
│status:pending                          │
╰────────────────────────────────────────╯
```
with
```
╭Delete Task─────────────────────────────╮
│[ID] [task]                             │
╰────────────────────────────────────────╯
```
and that the second `x` would delete the task.

**Comment by kdheepak** [comment]
Yes, it would be possible to implement it this way. I'll look into it :)

**Comment by kdheepak** [comment]
I've implemented this. I'll merge and update to a major release.

I've decided I would like to use `dd` or `d Enter` to delete the task. Previously, this was done by `x`. And previously marking something as `done` was done by `d`. With this PR, the key bindings will have changed. Hopefully, when user decide to mark something as `done`, the prompt will show them that this is now `delete`. In any case, users can undo by using `u`.

This will be less of a problem in the future when users will be able to set their own keybindings in a configuration. For now, I'm hoping users will put up with the slight inconvenience of a change in keybindings. 

It would be nice it I could announce to all users that the keybindings have changed.

**Comment by donbex** [comment]
Thanks for such a prompt response! However... I don't know how many users _taskwarrior-tui_ has at the moment, but are you sure the disruption of swapping the commands for `delete` and `done` is worth it?

If you really want to go that way, I think you should at least put a warning near the top of the readme. It might also be worth looking into some form of first-run disruption. For example, this could be some form of "pop-up" (like _midnight commander_'s confirmation screens) that needs to be acknowledged. Or, more simply, just print a notice to stdout and wait for acknowledgement before starting the tui. This should stay in place for at least a couple of versions.

If you choose to go down the pop-up route, it might be nice to make it into a brief "what's new" overlay window showing a brief changelog for the last couple of versions, and add an option to disable it.

**Comment by kdheepak** [comment]
After sleeping on it, yes I don't think the disruption of swapping is worth it. But I imagine there might be similar breaking changes in the future before a stable version is released. So yes, the "What's new?" pop up that has to be manually dismissed on first run seems to be the solution. I'll have to think about a little more (maybe over the weekend) before deciding whether to use `xx` to delete and `d` to mark as done instead for that PR. So I'll have to leave that open for a little longer.

**Comment by david-perez** [comment]
I'd prefer to keep `d` for `done` and `x` for `delete`, with a confirmation prompt on `x`.

**Comment by xarthurx** [comment]
I would say, allowing multiple shortcuts to bind on the same action would solve the problem of deciding which key to use....
For instance, VIM allows `ZZ` or `:wq` for exiting the program.

**Comment by vizcay** [comment]
I've using this tool for a month now and having a y/n confirmation prompt (maybe configurable for not disrupting current users) will be really great.

Accidentally marking as done a task or deleting it happens from time to time (and you don't always get to see it to trigger an undone!).

**Comment by dimas-cyriaco** [comment]
This would be very useful. I'm constantly deleting tasks while trying to press 'z' to hide the info panel... 😅

**Comment by kdheepak** [comment]
Thanks for everyone's patience on this! I've updated the PR with this feature:

https://user-images.githubusercontent.com/1813121/127206080-1e8af5a5-61d7-42c8-98e3-d666715d763d.mov

You by default you can use `x -> x` to "prompt and confirm delete" or `x -> q` to "prompt and abort delete". These keys are configurable. You can also use "Enter" or "Esc" to confirm or abort and that is not configurable but I think that is something I think people might find useful. For the sake of simplicity, I'm not adding `y` / `n` as confirm / abort options.

**Comment by kdheepak** [comment]
This feature is now available in https://github.com/kdheepak/taskwarrior-tui/releases/tag/v0.13.28.

**Comment by vizcay** [comment]
@kdheepak great work, already have it updated and testing!

Which are the config keys to parametrize the defaults x and q ? I've tried
```
uda.taskwarrior-tui.keyconfig.enter=y
uda.taskwarrior-tui.keyconfig.esc=n
```
without luck.

**Comment by kdheepak** [comment]
currently I’m using this: 

```
uda.taskwarrior-tui.keyconfig.delete=x
uda.taskwarrior-tui.keyconfig.quit=q 
```

but I think you are expecting a different key for confirmation, which is not currently supported. I can add that as a feature.
